redash_toolbelt/examples/migrate.py

```diff
@@ -0,0 +1,313 @@
+# Copied from https://gist.github.com/arikfr/2c7d09f6837b256c58a3d3ef6a97f61a
+
+import json
+import requests
+import logging
+import sys
+
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+logging.getLogger("requests").setLevel("ERROR")
+
+# The Redash instance you're copying from:
+ORIGIN = "https://redash.acme.com"
+ORIGIN_API_KEY = ""  # admin API key
+
+# The Redash account you're copying into:
+DESTINATION = 'https://app.redash.io/acme'
+DESTINATION_API_KEY = ""  # admin API key
+
+# You need to create the data sources in advance in the destination account. Once created, update the map here:
+# (origin Redash data source id -> destination Redash data source id)
+DATA_SOURCES = {
+    1: 1234,
+}
+
+
+meta = {
+    # include here any users you already created in the target Redash account.
+    # the key is the user id in the origin Redash instance. make sure to include the API key, as it used to recreate any objects
+    # this user might have owned.
+    "users": {
+        "1": {
+            "id": "",
+            "email": "",
+            "invite_link": "",
+            "api_key": ""
+        },
+    },
+    "queries": {},
+    "visualizations": {},
+    "dashboards": {}
+}
+
+
+def auth_headers(api_key):
+    return {
+        "Authorization": "Key {}".format(api_key)
+    }
+
+
+def api_request(api):
+    response = requests.get(ORIGIN + api, headers=auth_headers(ORIGIN_API_KEY))
+    response.raise_for_status()
+
+    return response.json()
+
+def get_paginated_resource(resource, api_key):
+    objects = []
+    headers = {'Authorization': 'Key {}'.format(api_key)}
+    has_more = True
+    page = 1
+    while has_more:
+        response = requests.get(resource, headers=headers,
+                                params={'page': page}).json()
+        objects.extend(response['results'])
+        has_more = page * response['page_size'] + 1 <= response['count']
+        page += 1
+
+    return objects 
+
+
+def get_queries(url, api_key):
+    path = "{}/api/queries".format(url)
+    return get_paginated_resource(path, api_key)
+
+
+def get_users(url, api_key):
+    path = "{}/api/users".format(url)
+    return get_paginated_resource(path, api_key)
+
+
+def import_users():
```

Comment:
This might fail due to rate limit (50/hr, 200/day). Worth at least documenting this if not add a backoff.

Reply:
I added a note to client.py about this. In the official instructions I'll recommend disabling rate limits for the migration.

Reply:
Is there a way to disable this, looks like it's hardcoded [here](https://github.com/getredash/redash/blob/d8d7c78992e44a4b6d7fdd4c39ccc1c35cd8c7a9/redash/handlers/users.py#L77).

As a side note, could it be useful to log the exception to help debug this issue?
```
        try:
            response = dest_client._post(f"api/users?no_invite=1", json=data)
        except Exception as ex:
            print(
                f"Could not create user: {ex}"
            )
            continue
```

Reply:
> Is there a way to disable this, looks like it's hardcoded [here](https://github.com/getredash/redash/blob/d8d7c78992e44a4b6d7fdd4c39ccc1c35cd8c7a9/redash/handlers/users.py#L77).

You can disable rate limits with [this setting](https://github.com/getredash/redash/blob/d8d7c78992e44a4b6d7fdd4c39ccc1c35cd8c7a9/redash/settings/__init__.py#L304).




> log the exception to help debug this issue?

This is a good idea. Will implement it.